python/pyspark/sql/connect/plan.py

```diff
@@ -343,6 +343,51 @@ def _repr_html_(self) -> str:
         """
 
 
+class Hint(LogicalPlan):
+    """Logical plan object for a Hint operation."""
+
+    def __init__(self, child: Optional["LogicalPlan"], name: str, params: List[Any]) -> None:
+        super().__init__(child)
+        self.name = name
+        self.params = params
+
+    def _convert_value(self, v: Any) -> proto.Expression.Literal:
+        value = proto.Expression.Literal()
+        if v is None:
+            value.null = True
+        elif isinstance(v, int):
+            value.integer = v
+        else:
+            value.string = v
+        return value
```

Comment:
This code has weird error behavior if v is not `None` or `int`. If for example, I were to assign a float, I would receive an error message from protobuf that is not actionable for the user. I think it would be good to either use the existing Python to Literal conversion code that we have or throw an exception.

Reply:
I improved the error handing logic here, and there are 4 occurrence of this similar logic, I'm planing to refactor it to reuse the existing code.